File: src/arduinobot_cpp_examples/src/simple_parameter.cpp
#include <rclcpp/rclcpp.hpp>
#include <string>
#include <rcl_interfaces/msg/set_parameters_result.hpp>
#include <memory>


using std::placeholders::_1;

class SimpleParameter : public rclcpp::Node
{
public:
    SimpleParameter() : Node("simple_parameter")
    {
        declare_parameter<int>("simple_int_param", 28);
        declare_parameter<std::string>("simple_string_param", "I'm Jebear");

        param_callback_handle_ = add_on_set_parameters_callback(std::bind(&SimpleParameter::paramChangedCallback, this, _1));
    }

private:
    OnSetParametersCallbackHandle::SharedPtr param_callback_handle_;

    rcl_interfaces::msg::SetParametersResult paramChangedCallback(const std::vector<rclcpp::Parameter> parameters)
    {
        rcl_interfaces::msg::SetParametersResult result;
        for (const auto &param : parameters)
        {
            if (param.get_name() == "simple_int_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
            {
                RCLCPP_INFO_STREAM(get_logger(), "simple_int_param new value " << param.as_int());
                result.successful = true;
            }
            if (param.get_name() == "simple_string_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_STRING)
            {
                RCLCPP_INFO_STREAM(get_logger(), "simple_stirng_param new value " << param.as_string());
                result.successful = true;
            }
        }
        return result;
    }
};

int main(int argc, char* argv[]){
    rclcpp::init(argc,argv);
    auto simpleParameter = std::make_shared<SimpleParameter>();
    rclcpp::spin(simpleParameter);
    rclcpp::shutdown();
    return 0;
}

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_parameter.py
import rclpy
from rclpy.node import Node
from rcl_interfaces.msg import SetParametersResult
from rclpy.parameter import Parameter

class SimpleParameter(Node):
    def __init__(self):
        super().__init__('simple_parameter')
        self.declare_parameter("simple_int_param", 28)
        self.declare_parameter("simple_string_param", "I'm Jebear")
        self.add_on_set_parameters_callback(self.paramChangeCallback)

    def paramChangedCallback(self, params):
        result = SetParametersResult()
        for param in params:
            if param.name == 'simple_int_param' and param.type_ == Parameter.Type.INTEGER:
                self.get_logger().info(f'simple_int_param value now {param.value}' )
                result.successful =True
            if param.name == 'simple_string_param' and param.type_ == Parameter.Type.STRING:
                self.get_logger().info(f'simple_string_param value now {param.value}' )
                result.successful =True
        return result

def main():
    rclpy.init()
    simpleParameter = SimpleParameter()
    rclpy.spin(simpleParameter)
    simpleParameter.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()